boot/bootutil/src/bootutil_public.c

```diff
@@ -548,7 +548,11 @@ boot_set_next(const struct flash_area *fa, bool active, bool confirm)
             /* The image slot is corrupt.  There is no way to recover, so erase the
              * slot to allow future upgrades.
              */
+#if !defined(CONFIG_MCUBOOT) && defined(__ZEPHYR__)
+            flash_area_flatten(fa, 0, flash_area_get_size(fa));
+#else
```

Comment:
think it was mentioned in a previous PR that OS specific code should not be present in here

Reply:
I can fix that although MCUboot has significant problem where we are coupled with Flash Map API which means that this has to be replaced with a function that is implemented for every supported system. I will figure way around.